src/shogun/machine/Machine.h

```diff
@@ -334,8 +334,40 @@ class CMachine : public CStoppableSGObject
 		 */
 		virtual bool train_machine(CFeatures* data=NULL)
 		{
-			SG_ERROR("train_machine is not yet implemented for %s!\n",
-					get_name());
+			bool result = false;
+			if (support_features_dispatching())
```

Comment:
This doesnt make sense: `train_machine` is overloaded by lots of shogun algorithms, so you cannot use that. You have to do the feature class dispatching in `train`, and it also needs to be optional (for the machines that are not CRTP), and the default option should be to just call `train_machine??`